vignettes/autoplieR.Rmd

```diff
@@ -196,32 +228,46 @@ ggplot(coefs_tidy, aes(x = LV, y = Weight, fill=Weight)) +
     scale_fill_distiller(type="div", palette="RdBu") 
 ```
 
-
-
 ## Extracting LVs and top pathways
 
-```{r LVs}
-# Retrieve LVs associated with being in the ICU
-top_LVs <- coefs_tidy %>% slice_max(order_by = abs(Weight), n = 5)
-autoPLIER.get_top_pathways(ap, LVs = top_LVs$LV, n_pathways = 10)
-
-# Retrieve LVs assocated with not being in the ICU
+Use the obtained coefficients from the previous section above to select significant LVs. In this example, positive coefficients corresponded to ICU admission while negative coefficients corresponded to non-ICU.
 
-# Retrieve top 10 pathways within those LVs associated with being in the ICU
-
-# Retrieve top 10 LVs associated with not being in the ICU
+```{r LVs, fig.dim = c(6, 6)}
+# Retrieve LVs associated with being in the ICU
+top_pos_LVs <- coefs_tidy %>% slice_max(order_by = Weight, n = 5)
+top_pos_LVs_pwys <- autoPLIER.get_top_pathways(ap, LVs = top_pos_LVs$LV, n_pathways = 10)
+
+top_pos_LVs_pwys %>%
+    bind_rows() %>%
+    mutate(LV = as.factor(LV)) %>%
+    ggplot(., aes(LV, pathway, size = value, color = value)) + 
+    geom_point() + 
+    guides(color=guide_legend(), size = guide_legend()) +
+    scale_color_gradient(low = "blue", high = "red") +
+    scale_size(range = c(0, 5)) +
+    theme_minimal(base_size = 8) +
+    ggtitle("Pathways corresponding to LVs associated with ICU")
+
+# Retrieve LVs associated with not being in the ICU
+top_neg_LVs <- coefs_tidy %>% slice_min(order_by = Weight, n = 5)
+top_neg_LVs_pwys <- autoPLIER.get_top_pathways(ap, LVs = top_neg_LVs$LV, n_pathways = 10)
+
+top_neg_LVs_pwys %>%
+    bind_rows() %>%
+    mutate(LV = as.factor(LV)) %>%
+    ggplot(., aes(LV, pathway, size = value, color = value)) + 
+    geom_point() + 
+    guides(color=guide_legend(), size = guide_legend()) +
+    scale_color_gradient(low = "blue", high = "red") +
+    scale_size(range = c(0, 5)) +
+    theme_minimal(base_size = 8) +
+    ggtitle("Pathways corresponding to LVs associated with non-ICU") 
```

Comment:
This is much better than your earlier visualizations. However, it would be better if you sorted the data prior to plotting.

Try using `arrange()` to sort the data. Once sorted, you can get the pathways to conform to the sorted data order by running this:

```R
... %>%
  mutate(pathway = factor(pathway, levels = unique(.$pathway))) %>%
  ...
```